modules/utils/include/hephaestus/utils/timing/rate_limiter.h

```diff
@@ -0,0 +1,31 @@
+//=================================================================================================
+// Copyright (C) 2023-2024 HEPHAESTUS Contributors
+//=================================================================================================
+
+#pragma once
+
+#include <chrono>
+#include <functional>
+
+namespace heph::utils::timing {
+
+/// @class RateLimiter
+/// @brief A class to limit the rate of function execution.
+///
+/// This class provides a mechanism to ensure that a function is not executed more frequently than a specified
+/// interval. Might be handy to use with logging.
+class RateLimiter final {
+  using ClockT = std::chrono::system_clock;
+
+public:
+  explicit RateLimiter(std::chrono::milliseconds rate);
```

Comment:
```suggestion
  explicit RateLimiter(std::chrono::milliseconds period);
```
rate is in Hertz (1/s), period is in (milli)seconds